File: packages/csharp/src/Anip.Service/CheckpointPolicy.cs
namespace Anip.Service;

/// <summary>
/// Configures automatic checkpoint creation.
/// </summary>
public class CheckpointPolicy
{
    /// <summary>How often to check for new entries (default 60 seconds).</summary>
    public int IntervalSeconds { get; set; } = 60;

    /// <summary>Minimum new entries before creating a checkpoint.</summary>
    public int MinEntries { get; set; } = 1;
}

File: packages/java/anip-service/src/main/java/dev/anip/service/CheckpointPolicy.java
package dev.anip.service;

/**
 * Configures automatic checkpoint creation.
 */
public class CheckpointPolicy {

    private int intervalSeconds = 60;
    private int minEntries = 1;

    public CheckpointPolicy() {}

    public CheckpointPolicy(int intervalSeconds, int minEntries) {
        this.intervalSeconds = intervalSeconds;
        this.minEntries = minEntries;
    }

    public int getIntervalSeconds() {
        return intervalSeconds;
    }

    public CheckpointPolicy setIntervalSeconds(int intervalSeconds) {
        this.intervalSeconds = intervalSeconds;
        return this;
    }

    public int getMinEntries() {
        return minEntries;
    }

    public CheckpointPolicy setMinEntries(int minEntries) {
        this.minEntries = minEntries;
        return this;
    }
}
